packages/tools/src/eventListeners/segmentation/segmentationDataModifiedEventListener.ts

```diff
@@ -101,7 +101,10 @@ const onSegmentationDataModified = function (
           const currentSegmentationImageId =
             imageIdReferenceMap.get(currentImageId);
 
-          const segmentationImage = cache.getImage(currentSegmentationImageId);
+          // Todo: fix the logic to support multiple derived imageIds
+          const segImageId = [...currentSegmentationImageId][0];
```

Comment:
This seems like a bad idea to convert to an array and back - it is relatively slow - you can instead get the iterator if you want and the first element, or add a helper function to do that for you.